pinot-core/src/main/java/org/apache/pinot/core/common/MinionConstants.java

```diff
@@ -136,4 +136,18 @@ public static class SegmentGenerationAndPushTask {
     public static final String CONFIG_NUMBER_CONCURRENT_TASKS_PER_INSTANCE =
         "SegmentGenerationAndPushTask.numConcurrentTasksPerInstance";
   }
+
+  public static class UpsertCompactionTask {
+    public static final String TASK_TYPE = "UpsertCompactionTask";
+    /**
+     * The time period to wait before picking segments for this task
+     * e.g. if set to "2d", no task will be scheduled for a time window younger than 2 days
+     */
+    public static final String BUFFER_TIME_PERIOD_KEY = "bufferTimePeriod";
+    /**
+     * The maximum percent of old records allowed for a completed segment.
+     * e.g. if the percent surpasses 30, then the segment will be compacted
+     */
+    public static final String INVALID_RECORDS_THRESHOLD_PERCENT = "invalidRecordsThresholdPercent";
```

Comment:
@Jackie-Jiang I agree with @snleee that representing this threshold as a percent is more user friendly. To eliminate the possibility of compacting small segments, the user could optionally configure a min segment size threshold (represented by num of total docs). WDYT?

Reply:
@Jackie-Jiang @robertzych 

In that case, we can have 2 configurations : num Rows + percent.

We can put the default numRows to be some reasonable number (e.g. 100k)

Then, the compaction would trigger when both condition satisfies.

Reply:
I've added a new config called minRecordCount (with a default of 100k) that will exclude smaller segments from compaction.

Reply:
@snleee @robertzych I still prefer using record count as the threshold so that each time when a segment is compacted, we know it will reduce enough records to justify the cost of the task. How about keeping both, and user can decide which one to use

Reply:
Sounds good. Then, we may need to change the logic to use only 1 config.

1. Pick `minRecordCount` configuration. If no configuration is provided, we use default value for minRecordCount.
2. If `thresholdPercent` is configured, we override the behavior to pick up `thresholdPercent`. If both record/percent are provided, we use percent.


@Jackie-Jiang Does this sound good to you?

Reply:
@Jackie-Jiang @snleee Some users may want to use both a percentage and a record count at the same time. For example, they may choose to start with a percentage then add a record count later to prevent compaction on smaller segments. Another benefit of using both concurrently may be to help prevent compaction in cases when the user provided a small value for the record count. In this case, the higher percentage would prevent compaction and therefore compensate for the small record count config. 

How does keeping percentage as required but adding record count as an optional config sound?

Reply:
@Jackie-Jiang The current changes allows the user to choose between percent, count, or both.